File: src/main/java/com/demonwav/mcdev/platform/mcp/at/psi/mixins/AtFieldNameMixin.java
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2017 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.mcp.at.psi.mixins;

import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;

public interface AtFieldNameMixin extends PsiElement {

    @NotNull
    PsiElement getNameElement();

    void setFieldName(@NotNull String fieldName);

    @NotNull
    String getFieldNameText();
}

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/mixins/AtFieldNameMixin.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2019 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.mcp.at.psi.mixins

import com.demonwav.mcdev.platform.mcp.at.psi.AtElement
import com.intellij.psi.PsiElement

interface AtFieldNameMixin : AtElement {

    val nameElement: PsiElement
    val fieldNameText: String

    fun setFieldName(fieldName: String)
}
